core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -683,6 +746,68 @@ class half_t {
     return __half2float(val) >= __half2float(rhs.val);
 #endif
   }
+
+  KOKKOS_FUNCTION
+  friend bool operator==(const volatile half_t& lhs,
+                         const volatile half_t& rhs) {
+    return static_cast<bool>(
+        reinterpret_cast<const volatile uint16_t&>(lhs.val) ==
+        reinterpret_cast<const volatile uint16_t&>(rhs.val));
+  }
+
+  KOKKOS_FUNCTION
+  friend bool operator!=(const volatile half_t& lhs,
+                         const volatile half_t& rhs) {
+    return static_cast<bool>(
+        reinterpret_cast<const volatile uint16_t&>(lhs.val) !=
+        reinterpret_cast<const volatile uint16_t&>(rhs.val));
+  }
+
+  KOKKOS_FUNCTION
+  friend bool operator<(const volatile half_t& lhs,
+                        const volatile half_t& rhs) {
+    return static_cast<bool>(
+        reinterpret_cast<const volatile uint16_t&>(lhs.val) <
+        reinterpret_cast<const volatile uint16_t&>(rhs.val));
```

Comment:
uhm this seems rather wrong. Does this actually hold true? I mean reinterpret cast the bits as uint16_t and then compare? Isn't like the sign bit set for negative?